test/apps/iodemo/io_demo.cc

```diff
@@ -1923,58 +1941,96 @@ class DemoClient : public P2pDemoCommon {
+                          // operations
+        long total; // Total number of completed operations
+    };
+
     void report_performance(long num_iters, double elapsed,
                             size_t total_bytes[IO_OP_MAX]) {
         if (num_iters == 0) {
             return;
         }
 
         double latency_usec = (elapsed / num_iters) * 1e6;
-        bool first_print    = true;
-
+        std::vector<io_op_perf_info_t> io_op_perf_info;
         UcxLog log(LOG_PREFIX);
 
-        for (unsigned op_id = 0; op_id < IO_OP_MAX; ++op_id) {
-            if (total_bytes[op_id] == 0) {
-                continue;
-            }
+        io_op_perf_info.resize(IO_OP_MAX + 1);
+        for (int op = 0; op <= IO_OP_MAX; ++op) {
+            io_op_perf_info[op].min       = std::numeric_limits<long>::max();
+            io_op_perf_info[op].max       = 0;
+            io_op_perf_info[op].min_index = _server_info.size();
+            io_op_perf_info[op].total     = 0;
+        }
 
-            if (!first_print) {
-                log << " | "; // print separator for non-first operation
-            }
-            first_print = false;
-
-            // Report bandwidth
-            double throughput_mbs = total_bytes[op_id] / UCS_MBYTE / elapsed;
-            log << io_op_names[op_id] << " " << throughput_mbs << " MBs";
-            total_bytes[op_id] = 0;
-
-            // Collect min/max among all connections
-            long delta_min = std::numeric_limits<long>::max(), delta_max = 0;
-            size_t min_index = 0, total_completed = 0;
-            for (size_t server_index = 0; server_index < _server_info.size();
-                 ++server_index) {
-                server_info_t& server_info = _server_info[server_index];
-                long delta_completed       = server_info.num_completed[op_id] -
-                                             server_info.prev_completed[op_id];
-                if ((delta_completed < delta_min) ||
-                    ((delta_completed == delta_min) &&
+        // Collect min/max among all connections
+        for (size_t server_index = 0; server_index < _server_info.size();
+             ++server_index) {
+            server_info_t& server_info = _server_info[server_index];
+            long total_completed       = 0;
+
+            for (int op = 0; op <= IO_OP_MAX; ++op) {
+                if (op == IO_OP_MAX) {
+                    // Find total minimum and maximum among all operations
+                    io_op_perf_info[IO_OP_MAX].min    =
+                            ucs_min(io_op_perf_info[IO_OP_MAX].min,
+                                    total_completed);
+                    io_op_perf_info[IO_OP_MAX].max    =
+                            ucs_max(io_op_perf_info[IO_OP_MAX].max,
+                                    total_completed);
+                    io_op_perf_info[IO_OP_MAX].total += total_completed;
+                    continue;
+                }
```

Comment:
it's code duplication with lines 2001-2005 (BTW why one uses ucs_min and other std::min?)
need to remove this code block and instead -- calculate delta_completed differently in line 1990: in case of OP_MAX it would be get_num_completed_round(READ) + get_num_completed_round(WRITE). OR - you can keep 'total_completed' var, and in case on OP_MAX set delta_completed=total_completed
```
if (op==MAX) {
  delta_completed = total_completed;
} else {
  total_completed += delta_completed;
}
```

Reply:
done